pkg/kubectl/versioned_client.go

```diff
@@ -18,28 +18,13 @@ package kubectl
 
 import (
 	externalclientset "k8s.io/client-go/kubernetes"
-	apps "k8s.io/client-go/kubernetes/typed/apps/v1beta1"
-	core "k8s.io/client-go/kubernetes/typed/core/v1"
-	extensions "k8s.io/client-go/kubernetes/typed/extensions/v1beta1"
 	internalclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
 )
 
-func versionedClientsetForDeployment(internalClient internalclientset.Interface) externalclientset.Interface {
+// TODO: get rid of this and plumb the caller correctly
+func versionedClientset(internalClient internalclientset.Interface) externalclientset.Interface {
 	if internalClient == nil {
 		return &externalclientset.Clientset{}
 	}
-	return &externalclientset.Clientset{
-		CoreV1Client:            core.New(internalClient.Core().RESTClient()),
-		ExtensionsV1beta1Client: extensions.New(internalClient.Extensions().RESTClient()),
-	}
-}
-
-func versionedClientsetForDaemonSet(internalClient internalclientset.Interface) externalclientset.Interface {
-	if internalClient == nil {
-		return &externalclientset.Clientset{}
-	}
-	return &externalclientset.Clientset{
-		AppsV1beta1Client:       apps.New(internalClient.Apps().RESTClient()),
-		ExtensionsV1beta1Client: extensions.New(internalClient.Extensions().RESTClient()),
-	}
+	return externalclientset.New(internalClient.Core().RESTClient())
```

Comment:
Good point. Fixed. I really disliked half-nil clientsets. That's bad practice. I have switched all the code using `versionedClientset` to explicit, versioned api group clientset interfaces. Much cleaner IMO now.